hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java

```diff
@@ -106,7 +108,11 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
     final OMRequest omRequest = super.preExecute(ozoneManager);
     // Verify name
     OmUtils.validateSnapshotName(snapshotName);
-
+    // Updating the volumeName & bucketName in case the bucket is a linked bucket. We need to do this before a
+    // permission check, since linked bucket permissions and source bucket permissions could be different.
+    ResolvedBucket bucket = ozoneManager.resolveBucketLink(Pair.of(volumeName, bucketName), this);
```

Comment:
Do we need allow snapshot operation over linked bucket?
As there may be case like any one can try create snapshot over both read bucket and linked bucket. We should restrict operation over linked bucket.

Reply:
There could be use cases where one might want to take a snapshot on a linked bucket. We can provide that option. The only downside is multiple linked buckets could have name collision post this.

Reply:
We can more opinion if really need support this, as it will increase un-necessary load without knowing actual information at backend (if its linked).

Reply:
That is not necessary case. Consider the case I have a use case for multiple snapshots to preserve a state of the multiple tables on separate instances and I intend to run some jobs on top of these snapshotted of tables at the same time. I intend to do take a snapshot on the planning stage of the job and intend to delete on cleanup once the job finishes

Reply:
> We can more opinion if really need support this, as it will increase un-necessary load without knowing actual information at backend (if its linked).
We can look into adding some info on the snapshot info something like is linked and the linkedBucketInfo volume & bucket info. It would act as just an info though for debug info.